components/availability/src/Availability.svelte

```diff
@@ -646,6 +723,189 @@
+        days.forEach((day) =>
+          day.slots
+            .filter((slot) => slot.selectionPending)
+            .forEach((slot) => (slot.selectionPending = false)),
+        );
+
+        draggedBlockSlots?.forEach((slot) => {
+          let slotPlusDelta =
+            day.slots[dragStartDay!.slots.indexOf(slot) + delta];
+          if (slotPlusDelta) {
+            slotPlusDelta.selectionPending = true;
+          }
+        });
+
+        days = [...days]; // re-render
+      } else {
+        // When drag-creating an event, dragging downward has different behaviour than dragging upward
+        // especially when you consider max_bookable_slots
+        let direction: "forward" | "backward" = "forward";
+
+        if (allow_booking && day === dragStartDay && dragStartSlot) {
+          if (slot.start_time < dragStartSlot.start_time) {
+            direction = "backward";
+          }
+
+          // Mark a given slot as selectionPending=true if it falls between where we started dragging, and where we're dragging now
+          day.slots.forEach((daySlot) => {
+            if (
+              direction === "forward"
+                ? daySlot.start_time >= dragStartSlot!.start_time &&
+                  daySlot.start_time <= slot.start_time
+                : daySlot.start_time <= dragStartSlot!.start_time &&
+                  daySlot.start_time >= slot.start_time
+            ) {
+              daySlot.selectionPending =
+                daySlot.availability !== AvailabilityStatus.BUSY ? true : false; // when you're dragging over a busy spot, don't set it to pending.
+            } else {
+              if (daySlot.selectionPending) {
+                daySlot.selectionPending = false;
+              }
+            }
+          });
+
+          // Don't let the user book more slots than are allowed
+          let tooManySlots =
+            day.slots.filter(
+              (s) =>
+                s.selectionPending ||
+                s.selectionStatus === SelectionStatus.SELECTED,
+            ).length > max_bookable_slots;
+          let pendingSlots = day.slots.filter((x) => x.selectionPending);
+          if (tooManySlots) {
+            let allowableSlots =
+              max_bookable_slots -
+              day.slots.filter(
+                (x) => x.selectionStatus === SelectionStatus.SELECTED,
+              ).length;
+            if (direction === "forward") {
+              // Only select the first N allowed slots AFTER your initially-dragegd one
+              pendingSlots
+                .slice(allowableSlots)
+                .forEach((slot) => (slot.selectionPending = false));
+            } else {
+              // Only select the first N allowed slots BEFORE your initially-dragegd one
+              pendingSlots
+                .slice(0, -allowableSlots)
+                .forEach((slot) => (slot.selectionPending = false));
+            }
+          }
+        }
+        days = [...days]; // re-render
```

Comment:
if `days = [...days];` is necessary, you could move it to the next line because it should execute for both the `if` and `else` blocks

Reply:
Good suggestion!